### ISSUE TITLE
Wrong output of a series expansion at infinite

### ISSUE THREAD

**Issue by edeustua** [sympy/sympy#20106]
The series of `g = sqrt(1/(r**2 + eta**(-2)))` at the infinite, where `r` and `eta` are symbols, produces wrong coefficients on higher order terms.

The output of `sresult = series(g, r, oo, n=10)` is `3/(128*eta**8*r**9) + 3/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))` differs from the correct answer `35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r +
O(r**(-10), (r, oo))` given by `g.aseries(x=r, n=10)`. These findings have been compared against results from Mathematica and Maxima.

Also, this issue disappears when one modifies `g = 1/sqrt(r**2 + eta**(-2))`.

I'm currently using python 3.8.5 with sympy 1.6.2, but have seen the issue happening on sympy live as well.

A script that examplifies this behavior is:
```python
from sympy import series, symbols, oo, sqrt

r, eta = symbols('r eta')
g = sqrt(1/ (r**2 + eta**(-2)))
gm = 1/ sqrt(r**2 + eta**(-2))

series_result = series(g, r, oo, n=10)
aseries_result = g.aseries(r, n=10)
print(series_result)
# wrong expression -> 3/(128*eta**8*r**9) + 3/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
print(aseries_result)
# 35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))

print('------------')

series_result = series(gm, r, oo, n=10)
aseries_result = gm.aseries(r, n=10)
print(series_result)
# 35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
print(aseries_result)
# 35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
```

**Comment by 0sidharth** [comment]
This has been fixed from sympy 1.7 onwards. On current master - 
```py
In [1]: from sympy import series, symbols, oo, sqrt
   ...: 
   ...: r, eta = symbols('r eta')
   ...: g = sqrt(1/ (r**2 + eta**(-2)))
   ...: gm = 1/ sqrt(r**2 + eta**(-2))
   ...: 
   ...: series_result = series(g, r, oo, n=10)
   ...: aseries_result = g.aseries(r, n=10)
   ...: print(series_result)
   ...: print(aseries_result)
   ...: 
   ...: print('------------')
   ...: 
   ...: series_result = series(gm, r, oo, n=10)
   ...: aseries_result = gm.aseries(r, n=10)
   ...: print(series_result)
   ...: print(aseries_result)
35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
------------
35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
35/(128*eta**8*r**9) - 5/(16*eta**6*r**7) + 3/(8*eta**4*r**5) - 1/(2*eta**2*r**3) + 1/r + O(r**(-10), (r, oo))
```

**Comment by edeustua** [comment]
As reported by @0sidharth, this is now fixed. :)
